server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/reactivestreams/SubscriberInputStream.scala

```diff
@@ -0,0 +1,136 @@
+package sttp.tapir.server.netty.internal.reactivestreams
+
+import io.netty.buffer.ByteBuf
+import io.netty.handler.codec.http.HttpContent
+import org.reactivestreams.{Publisher, Subscriber, Subscription}
+import sttp.capabilities.StreamMaxLengthExceededException
+
+import java.io.{IOException, InputStream}
+import java.util.concurrent.LinkedBlockingQueue
+import scala.annotation.tailrec
+import scala.concurrent.Promise
+
+/** A blocking input stream that reads from a reactive streams publisher of [[HttpContent]].
+  * @param maxBufferedChunks
+  *   maximum number of unread chunks that can be buffered before blocking the publisher
+  */
+private[netty] class SubscriberInputStream(maxBufferedChunks: Int = 1) 
+  extends InputStream with Subscriber[HttpContent] {
+  
+  require(maxBufferedChunks > 0)
+
+  import SubscriberInputStream._
+
+  // volatile because used in both InputStream & Subscriber methods
+  @volatile private[this] var closed = false
+  // Calls on the subscription must be synchronized in order to satisfy the Reactive Streams spec
+  // (https://github.com/reactive-streams/reactive-streams-jvm?tab=readme-ov-file#2-subscriber-code - rule 7)
+  // because they are called both from InputStream & Subscriber methods.
+  private[this] var subscription: Subscription = _
+  private[this] var currentItem: Item = _
+  // the queue serves as a buffer to allow for possible parallelism between the subscriber and the publisher
+  private val queue = new LinkedBlockingQueue[Item](maxBufferedChunks + 1) // +1 to have a spot for End/Error
+
+  private def readItem(blocking: Boolean): Item = {
+    if (currentItem eq null) {
+      currentItem = if (blocking) queue.take() else queue.poll()
+      currentItem match {
+        case _: Chunk => synchronized(subscription.request(1))
```

Comment:
to run on loom, we should avoid synchronized (causes thread pinning) and use a lock instead